nomos-sdp/src/lib.rs

```diff
@@ -35,9 +36,14 @@ impl Locator {
     pub const fn new(addr: Multiaddr) -> Self {
         Self { addr }
     }
+
+    #[must_use]
+    pub const fn addr(&self) -> &Multiaddr {
+        &self.addr
+    }
 }
 
-#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
```

Comment:
In this case we should use some `repr(c)` or `repr(u8)`, this is gonna go through wire right?

Reply:
The `nomos-sdp` is core library and it should contain serialization for over the wire comms. If we go this path, then I'd keep definitions here as per spec, this is defined this way in spec for future expansion. The [Mantle PR](https://github.com/logos-co/nomos/pull/1310) already makes the assumption that SDP lib will implement the serialization, maybe we could just revert this change for now if it's not critical for this PR?